### PR TITLE
Add HTTP status code 418

### PR DESCRIPTION
**Pull request by ametrocavich**
Added support for status code 418 according to https://tools.ietf.org/html/rfc2324

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @ametrocavich,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by ametrocavich** [comment]
I signed the CLA

**Comment by jlprat** [comment]
OK TO TEST

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala [hunk 1]
```diff
@@ -142,6 +142,7 @@ object StatusCodes extends ObjectRegistry[Int, StatusCode] {
   val UnsupportedMediaType         = reg(c(415)("Unsupported Media Type", "The request entity has a media type which the server or resource does not support."))
   val RequestedRangeNotSatisfiable = reg(c(416)("Requested Range Not Satisfiable", "The client has asked for a portion of the file, but the server cannot supply that portion."))
   val ExpectationFailed            = reg(c(417)("Expectation Failed", "The server cannot meet the requirements of the Expect request-header field."))
+  val ImATeapot                    = reg(c(418)("I'm a teapot", "The resulting entity body MAY be short and stout."))
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala]: If this is seriously meant to get in then the PR is missing Java API ;-)
You'd have to mirror it in `javadsl/model/StatusCodes`

Comment (ametrocavich) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala]: Done :)

**Comment by jrudolph** [comment]
Great first addition, @ametrocavich ;) :coffee: :tea:

**Comment by btomala** [comment]
This status is missing in the 10.1.1 version. Is it regression or deliberate action?

**Comment by ktoso** [comment]
Please be specific. Missing where? It’s srill present in sources: https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala

**Pull request merged**